src/anyp/PortCfg.cc

```diff
@@ -25,13 +29,40 @@ AnyP::PortCfgPointer FtpPortList;
 int NHttpSockets = 0;
 int HttpSockets[MAXTCPLISTENPORTS];
 
-AnyP::PortCfg::PortCfg(const TrafficModeFlags::PortKind aPortKind):
+static AnyP::TrafficModeFlags::PortKind
+ParseDirective(const SBuf &directive)
+{
+    if (directive.cmp("http_port") == 0)
+        return AnyP::TrafficModeFlags::httpPort;
+    else if (directive.cmp("https_port") == 0)
+        return AnyP::TrafficModeFlags::httpsPort;
+    else {
+        assert(directive.cmp("ftp_port") == 0);
+        return AnyP::TrafficModeFlags::ftpPort;
+    }
+}
+
+static AnyP::ProtocolVersion
+DefaultProtocol(const SBuf &directive)
+{
+    if (directive.cmp("http_port") == 0)
+        return Http::ProtocolVersion(1,1);
+    else if (directive.cmp("https_port") == 0)
+        return AnyP::ProtocolVersion(AnyP::PROTO_HTTPS, 1,1);
+    else {
+        assert(directive.cmp("ftp_port") == 0);
+        return Ftp::ProtocolVersion();
+    }
+}
+
+AnyP::PortCfg::PortCfg(const SBuf &directive):
     next(),
     s(),
-    transport(AnyP::PROTO_HTTP,1,1), // "Squid is an HTTP proxy", etc.
+    directiveName(directive),
+    transport(DefaultProtocol(directive)), // "Squid is an HTTP proxy", etc.
```

Comment:
We no longer need to (pretend to) justify that hard-coded default.

```suggestion
    transport(DefaultProtocol(directive)),
```

Reply:
Done.